app/services/ipam/db.rb

```diff
@@ -8,7 +8,7 @@ def suggest_ip
       to = subnet.to.present? ? IPAddr.new(subnet.to) : IPAddr.new(subnet_range.last.to_i - 2, subnet.family)
       (from..to).each do |address|
         ip = address.to_s
-        if !subnet.known_ips.include?(ip) && !excluded_ips.include?(ip)
+        if !subnet.known_ips.include?(ip) && !(excluded_ips || []).include?(ip)
```

Comment:
Well, what you are trying to prevent is the following:

```ruby
> {:abc => nil}.fetch(:abc, [])
 => nil
```

Fetch does not help you here. It just takes the provided default it the key cannot be found in the hash.


I do think, defining a custom getter is overengineering the whole thing. But that might just be a matter of taste.